beacon-chain/blockchain/receive_attestation.go

```diff
@@ -182,9 +182,14 @@ func (s *Service) notifyEngineIfChangedHead(ctx context.Context, newHeadRoot [32
 		log.WithError(errNilFinalizedInStore).Error("could not get finalized checkpoint")
 		return
 	}
-	newHeadBlock, err := s.cfg.BeaconDB.Block(ctx, newHeadRoot)
+	if err := s.cfg.BeaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
```

Comment:
if we pass in 0 blocks, do we still write to the db ? I think `SaveBlocks` should exit early if there is nothing to write.

Reply:
I think I will just remove it. This was from the previous solution. Now `getBlock` should take care of it. Doesn't seem right to Save cached blocks here